tensorflow/compiler/tests/image_ops_test.py

```diff
@@ -514,6 +514,24 @@ def testAlignCorners3x3To12x12(self):
                            [7, 7, 7, 8, 8, 8, 8, 8, 8, 9, 9, 9]],
                           dtype=np.float32))
 
+  def testAlignCorners3x3To12x12_uint8(self):
```

Comment:
Would you please explain how this test relates to the heuristic change?

Reply:
Sorry, this test change was for a different PR https://github.com/tensorflow/tensorflow/pull/31156.  I will revert it in this branch.

Reply:
Please ignore my previous comment on this change.  I thought that it mixed into a different branch.

Regarding https://github.com/tensorflow/tensorflow/pull/31156, this test ensures that integer-type nearest neighbor resizing proceeds without any error.  It would trigger error for GPU without the heuristic change in https://github.com/tensorflow/tensorflow/commit/fd34d066ec5f514d737d771efaf0f6cf93925cf7#diff-e46056ad1b86847e16150ab7906bab3b.  According to https://github.com/tensorflow/tensorflow/pull/30336#discussion_r303168563, CPU backend would also fail.